com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -74,7 +89,8 @@ public int Find(VariableKey key)
             {
                 if (Entries[i].Key.NetworkObjectId == key.NetworkObjectId &&
                     Entries[i].Key.BehaviourIndex == key.BehaviourIndex &&
-                    Entries[i].Key.VariableIndex == key.VariableIndex)
+                    Entries[i].Key.VariableIndex == key.VariableIndex &&
```

Comment:
Seems like it would be pretty straightforward to establish a dictionary and then de-ref the key.

On m_tickIndex, I had to stop to reason about this code.  I think I would name it m_TickAsIndex else my brain really wants it to mean "the index of a tick"

Reply:
Ok, so this logic is, if I find a matching entry AND (either I'm not using tick as an index OR the tick written matches the one I'm looking at).

I'm assuming the tick matching part is so that we don't have to clear out old entries?  I get that.  But I'm not sure why if this isn't true that it's still ok for it to be a hit if m_tickIndex is false.

Reply:
In this (old) branch, we still had different Snapshots per connection. I was swayed, in our discussions, by Luke's very good and valid arguments. As such, the latest branch (spawn) doesn't need this anymore.

But to clarify _this_ code, in order for a Key to match, the `NetworkObjectId`, the `BehaviourIndex` and the `VariableIndex` have to match. Also, you need one of two things to hold:
- either this Snapshot doesn't care about ticks (`!m_TickIndex`) or
- the ticks actually match `(Entries[i].Key.TickWritten == key.TickWritten)`

This was so that per-connection snapshots would only maintain the latest value of each variable, but the main snapshot would maintain multiple.

I could comment a bit to improve, but since the next branch will drop this, it would be a bit moot.